pkg/signup/service/signup_service_test.go

```diff
@@ -1665,74 +1683,54 @@ func (s *TestSignupServiceSuite) TestGetDefaultUserNamespace() {
 	// given
 	s.ServiceConfiguration(configuration.Namespace(), true, "", 5)
 
-	signup := signup.Signup{
-		Name:              "dave#123",
-		CompliantUsername: "dave",
-	}
-
-	space := s.newSpaceForMUR(signup.CompliantUsername, signup.Name)
+	space := s.newSpace("dave")
 	err := s.FakeSpaceClient.Tracker.Add(space)
 	require.NoError(s.T(), err)
 
-	spacebinding := s.newSpaceBinding(signup.CompliantUsername, space.Name)
-	err = s.FakeSpaceBindingClient.Tracker.Add(spacebinding)
-	require.NoError(s.T(), err)
-
 	// when
-	defaultUserNamespace := service.GetDefaultUserNamespace(s, signup)
+	targetCluster, defaultUserNamespace := service.GetDefaultUserTarget(s, "dave", "dave")
 
 	// then
 	assert.Equal(s.T(), "dave-dev", defaultUserNamespace)
+	assert.Equal(s.T(), "member-123", targetCluster)
 
 	s.T().Run("informer", func(t *testing.T) {
 		// given
 		inf := fake.NewFakeInformer()
 		inf.GetSpaceFunc = func(_ string) (*toolchainv1alpha1.Space, error) {
 			return space, nil
 		}
-		inf.ListSpaceBindingFunc = func(_ ...labels.Requirement) ([]toolchainv1alpha1.SpaceBinding, error) {
-			return []toolchainv1alpha1.SpaceBinding{*spacebinding}, nil
-		}
 
 		// when
-		defaultUserNamespace := service.GetDefaultUserNamespace(inf, signup)
+		targetCluster, defaultUserNamespace := service.GetDefaultUserTarget(inf, "dave", "dave")
 
 		// then
 		assert.Equal(t, "dave-dev", defaultUserNamespace)
+		assert.Equal(s.T(), "member-123", targetCluster)
 	})
 }
 
-// TestGetDefaultUserNamespaceOnlyUnownedSpace tests that the default user namespace is returned even if the only accessible Space was not created by the user.
-// This is valuable because if the creator label on a Space is missing for whatever reason, APIs that depend on this field will still work.
+// TestGetDefaultUserNamespaceOnlyUnownedSpace tests that the default user namespace is returned even if the only accessible Space was not created as the home space.
+// This is valuable when user doesn't have default home space created, but has access to some shared spaces
```

Comment:
should we add a similar test, but with multipe spacebindings, just to make sure we always return the first one ? 

something like
```
func (s *TestSignupServiceSuite) TestGetDefaultUserNamespaceOnlyMultipleUnownedSpace() {
	// given
	s.ServiceConfiguration(configuration.Namespace(), true, "", 5)
	// space created for userA
	space := s.newSpace("userA")
	err := s.FakeSpaceClient.Tracker.Add(space)
	require.NoError(s.T(), err)

	spacebinding := s.newSpaceBinding("userB", space.Name)
	err = s.FakeSpaceBindingClient.Tracker.Add(spacebinding)
	require.NoError(s.T(), err)

	// space created for userC
	spaceC := s.newSpace("userC")
	err = s.FakeSpaceClient.Tracker.Add(spaceC)
	require.NoError(s.T(), err)

	// space shared with userB
	spacebindingC := s.newSpaceBinding("userB", spaceC.Name)
	err = s.FakeSpaceBindingClient.Tracker.Add(spacebindingC)
	require.NoError(s.T(), err)

	// when
	targetCluster, defaultUserNamespace := service.GetDefaultUserTarget(s, "", "userB")

	// then
	assert.Equal(s.T(), "userA-dev", defaultUserNamespace)
	assert.Equal(s.T(), "member-123", targetCluster)

	s.T().Run("informer", func(t *testing.T) {
		// given
		inf := fake.NewFakeInformer()
		inf.GetSpaceFunc = func(_ string) (*toolchainv1alpha1.Space, error) {
			return space, nil
		}
		inf.ListSpaceBindingFunc = func(_ ...labels.Requirement) ([]toolchainv1alpha1.SpaceBinding, error) {
			return []toolchainv1alpha1.SpaceBinding{*spacebinding}, nil
		}

		// when
		targetCluster, defaultUserNamespace := service.GetDefaultUserTarget(inf, "", "userB")

		// then
		assert.Equal(t, "userA-dev", defaultUserNamespace)
		assert.Equal(s.T(), "member-123", targetCluster)
	})
}
```

Reply:
makes sense indeed! Good point!